Lib/compression/zstd/__init__.py

```diff
@@ -0,0 +1,293 @@
+    """
+    # Check argument's type
+    if not isinstance(dict_size, int):
+        raise TypeError('dict_size argument should be an int object.')
+
+    # Prepare data
+    chunks = []
+    chunk_sizes = []
+    for chunk in samples:
+        chunks.append(chunk)
+        chunk_sizes.append(_nbytes(chunk))
+
+    chunks = b''.join(chunks)
+    if not chunks:
+        raise ValueError("The samples are empty content, can't train dictionary.")
+
+    # samples_bytes: samples be stored concatenated in a single flat buffer.
+    # samples_size_list: a list of each sample's size.
+    # dict_size: size of the dictionary, in bytes.
+    dict_content = _train_dict(chunks, chunk_sizes, dict_size)
+
+    return ZstdDict(dict_content)
+
+
+def finalize_dict(zstd_dict, samples, dict_size, level):
+    """Finalize a zstd dictionary, return a ZstdDict object.
+
+    Given a custom content as a basis for dictionary, and a set of samples,
+    finalize dictionary by adding headers and statistics according to the zstd
+    dictionary format.
+
+    You may compose an effective dictionary content by hand, which is used as
+    basis dictionary, and use some samples to finalize a dictionary. The basis
+    dictionary can be a "raw content" dictionary, see is_raw parameter in
+    ZstdDict.__init__ method.
+
+    Parameters
+    zstd_dict: A ZstdDict object, basis dictionary.
+    samples:   An iterable of samples, a sample is a bytes-like object
+               represents a file.
+    dict_size: The dictionary's maximum size, in bytes.
+    level:     The compression level expected to use in production. The
+               statistics for each compression level differ, so tuning the
+               dictionary for the compression level can help quite a bit.
+    """
+
+    # Check arguments' type
+    if not isinstance(zstd_dict, ZstdDict):
+        raise TypeError('zstd_dict argument should be a ZstdDict object.')
+    if not isinstance(dict_size, int):
+        raise TypeError('dict_size argument should be an int object.')
+    if not isinstance(level, int):
+        raise TypeError('level argument should be an int object.')
+
+    # Prepare data
+    chunks = []
+    chunk_sizes = []
+    for chunk in samples:
+        chunks.append(chunk)
+        chunk_sizes.append(_nbytes(chunk))
+
+    chunks = b''.join(chunks)
+    if not chunks:
+        raise ValueError("The samples are empty content, can't finalize dictionary.")
+
+    # custom_dict_bytes: existing dictionary.
+    # samples_bytes: samples be stored concatenated in a single flat buffer.
+    # samples_size_list: a list of each sample's size.
+    # dict_size: maximal size of the dictionary, in bytes.
+    # compression_level: compression level expected to use in production.
+    dict_content = _finalize_dict(zstd_dict.dict_content,
+                                  chunks, chunk_sizes,
+                                  dict_size, level)
+
+    return _zstd.ZstdDict(dict_content)
```

Comment:
We should be consistent about using either ZstdDict or prefixing with `_zstd`
```suggestion
    return ZstdDict(dict_content)
```